src/utils.rs

```diff
@@ -0,0 +1,280 @@
+use crate::error::SlateDBError;
+use crate::error::SlateDBError::{BackgroundTaskFailed, BackgroundTaskShutdown};
+use std::future::Future;
+
+pub(crate) struct WriteOnceRegister<T: Clone> {
```

Comment:
nit: how about naming it as something like `OnceWatchCell` ?

i suspect that Rust uses more `xxxCell` as an informal naming convention for what we call `Register` in other languages 🤔

Reply:
sounds good to me. will do